### PR TITLE
Feature 7 testing 

### PR DESCRIPTION
**Pull request by villejuutila**
Both, JVM and Native tests implemented. 

Also contains other changes related to for example building a native image.

New TimeEntryAPI and TimeEntryController was suggested by @jarinykanen.
Mainly to be able to clean resources persisted by tests, since earlier there was no endpoint for deleting persisted entries.
Also it was considered future proofing since at some point we probably want to be able to delete persisted entries.

### CODE REVIEW

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/SynchronizeTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * Tests /v1/synchronize -endpoint with 2000 random generated ForecastTimeEntries
+     */
+    @Test
+    fun testSynchronizationLoop() {
+        setScenario(
+            scenario = TIMES_SCENARIO,
+            state = GENERATED_STATE
+        )
+        createTestBuilder().use { testBuilder ->
+            val synchronized = testBuilder.manager.synchronization.synchronizeEntries()
```
Comment (anttileppa) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/SynchronizeTest.kt:83]: Is it possible that this test leaves from garbage behind?

File: src/main/kotlin/fi/metatavu/timebank/api/forecast/models/ForecastPerson.kt [hunk 2]
```diff
@@ -1,33 +1,41 @@
...
+    var startDate: String = ""
+    @JsonProperty("end_date")
+    var endDate: String? = null
+    @JsonProperty("created_at")
+    var createdAt: String = ""
+    @JsonProperty("updated_at")
+    var updatedAt: String? = null
+    @JsonProperty("is_system_user")
+    var isSystemUser: Boolean = false
+
```
Comment (jarinykanen) [inline src/main/kotlin/fi/metatavu/timebank/api/forecast/models/ForecastPerson.kt:40]:
```suggestion
```

File: src/main/kotlin/fi/metatavu/timebank/api/persistence/model/TimeEntry.kt [hunk 3]
```diff
@@ -43,13 +42,13 @@ class TimeEntry {
...
-
-        return differentFields.size == 1
+        if (this === timeEntry) return true
+        return forecastId == timeEntry.forecastId &&
+                person == timeEntry.person &&
+                internalTime == timeEntry.internalTime &&
+                projectTime == timeEntry.projectTime &&
+                date == timeEntry.date &&
+                createdAt == timeEntry.createdAt &&
+                updatedAt == timeEntry.updatedAt
```
Comment (jarinykanen) [inline src/main/kotlin/fi/metatavu/timebank/api/persistence/model/TimeEntry.kt:52]:
```suggestion
        return forecastId == timeEntry.forecastId &&
            person == timeEntry.person &&
            internalTime == timeEntry.internalTime &&
            projectTime == timeEntry.projectTime &&
            date == timeEntry.date &&
            createdAt == timeEntry.createdAt &&
            updatedAt == timeEntry.updatedAt
```

File: src/native-test/kotlin/fi/metatavu/timebank/api/test/NativeSystemTestIT.kt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package fi.metatavu.timebank.api.test
+
+import fi.metatavu.timebank.api.test.functional.resources.TestMySQLResource
+import fi.metatavu.timebank.api.test.functional.resources.TestWiremockResource
+import fi.metatavu.timebank.api.test.functional.tests.SystemTest
+import io.quarkus.test.common.QuarkusTestResource
+import io.quarkus.test.junit.QuarkusIntegrationTest
+
+
```
Comment (jarinykanen) [inline src/native-test/kotlin/fi/metatavu/timebank/api/test/NativeSystemTestIT.kt:9]:
```suggestion
```

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestData.kt [hunk 5]
```diff
@@ -19,12 +90,41 @@ class TestData {
...
+            holidayA.id = 1
+            holidayA.holidayCalendarId = 2
+            holidayA.year = 2022
+            holidayA.month = 6
+            holidayA.day = 5
+            holidayA.name = "Helluntai"
+            return listOf(holidayA)
+        }
+
+        private fun createForecastTimeEntryResponse(
```
Comment (jarinykanen) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestData.kt:117]: Add docs

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestPersonsData.kt [hunk 6]
```diff
@@ -113,22 +77,71 @@ class TestPersonsData {
...
+                    holidayCalendarId = 123456,
+                    startDate = "2022-06-30",
+                    createdAt = "2022-06-30",
+                    isSystemUser = false,
                 )
             )
+
+        }
+
+        private fun createTestPerson(
```
Comment (jarinykanen) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestPersonsData.kt:108]: Add docs

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+import fi.metatavu.timebank.api.forecast.models.ForecastTimeEntry
+
+/**
+ * Class for test time entries mock data
+ */
+class TestTimeEntriesData {
+
+    companion object {
+
+        fun getForecastTimeEntries(): List<ForecastTimeEntry> {
```
Comment (jarinykanen) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt:12]: Add docs, check all methods

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/PersonsTest.kt [hunk 8]
```diff
@@ -19,10 +19,9 @@ import org.junit.jupiter.api.Assertions.*
     QuarkusTestResource(TestMySQLResource::class),
     QuarkusTestResource(TestWiremockResource::class)
 )
-@TestProfile(LocalTestProfile::class)
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class PersonsTest: AbstractTest() {
-
+
```
Comment (jarinykanen) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/PersonsTest.kt:24]:
```suggestion

```

**Review by jarinykanen — CHANGES_REQUESTED** [review]

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/AbstractTest.kt [hunk 9]
```diff
@@ -27,13 +63,13 @@ abstract class AbstractTest {
             OkHttpClient()
                 .newCall(
                     Request.Builder()
-                    .url("$forecastBaseUrl/__admin/scenarios/reset")
+                    .url("http://localhost:8082/__admin/scenarios/reset")
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/AbstractTest.kt]: This particular variable is a config property overridden by our WireMock instance. 
For some strange reason config properties overridden by services other than Quarkus DevServices or Testcontainers do not get into the context of these test classes IN NATIVE MODE. In JVM mode it works just fine.
As of now, replaced this particular line with a variable, where the value is hard-coded tho.

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/AbstractTest.kt [hunk 10]
```diff
@@ -45,7 +81,7 @@ abstract class AbstractTest {
             OkHttpClient()
                 .newCall(
                     Request.Builder()
-                        .url("$forecastBaseUrl/__admin/scenarios/$scenario/state")
+                        .url("http://localhost:8082/__admin/scenarios/$scenario/state")
```
[the remark this answers is not in the document]
Comment (villejuutila) [reply src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/AbstractTest.kt]: This particular variable is a config property overridden by our WireMock instance. 
For some strange reason config properties overridden by services other than Quarkus DevServices or Testcontainers do not get into the context of these test classes IN NATIVE MODE. In JVM mode it works just fine.
As of now, replaced this particular line with a variable, where the value is hard-coded tho.

**Review by jarinykanen — APPROVED** [review]

**Pull request merged**